### PR TITLE
i18n(es): Translated `content-collections.mdx`

### PR DESCRIPTION
**Pull request by dreyfus92**
#### What kind of changes does this PR include?

- Translated content

#### Description

- What does this PR change? Give us a brief description.
Translated `content-collections.mdx` to spanish 🇪🇸

### CODE REVIEW

**Comment by dreyfus92** [comment]
The error logged by the GitHub Action is refering to the link in the line 750 `<RecipeLinks slugs={["en/recipes/reading-time" ]}/>` it won't go away cause we haven't translated the recipes to spanish yet.

File: src/content/docs/es/guides/content-collections.mdx [hunk 1]
```diff
@@ -0,0 +1,753 @@
...
+}
+```
+
+### Definiendo un esquema de colección
+
+Los esquemas hacen cumplir un frontmatter o datos de entrada consistentes dentro de una colección. Un esquema **garantiza** que estos datos existen en una forma predecible cuando necesitas hacer referencia o consultarlos. Si algún archivo viola su esquema de colección, Astro proporcionará un error útil para informarte. 
+
+Los esquemas también alimentan las escrituras automáticas de TypeScript de Astro para tu contenido. Cuando defines un esquema para tu colección, Astro generará y aplicará automáticamente una interfaz de TypeScript. El resultado es un soporte completo de TypeScript cuando consultas tu colección, incluyendo el autocompletado de propiedades y la comprobación de tipos.
+
+Para definir tu primer colleción, crea un archivo `src/content/config.ts` si no existe (las extensiones `.js` y `.mjs` también son compatibles). Este archivo debe:
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:143]:
```suggestion
Para definir tu primera colleción, crea un archivo `src/content/config.ts` si no existe (las extensiones `.js` y `.mjs` también son compatibles). Este archivo debe:
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 2]
```diff
@@ -0,0 +1,753 @@
...
+
+1. **Importa las utilidades adecuadas** de `astro:content`.
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:145]:
```suggestion
1. **Importar las utilidades adecuadas** de `astro:content`.
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 3]
```diff
@@ -0,0 +1,753 @@
...
+2. **Define cada colección que deseas validar.** Esto incluye un `type` (introducido en Astro v2.5.0) que especifica si la colección contiene formatos de autoría de contenido como Markdown (`type: 'content'`) o formatos de datos como JSON o YAML (`type: 'data'`). También incluye un `schema` que define la forma de tu frontmatter o datos de entrada.
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:146]:
```suggestion
2. **Definir cada colección que deseas validar.** Esto incluye un `type` (introducido en Astro v2.5.0) que especifica si la colección contiene formatos de autoría de contenido como Markdown (`type: 'content'`) o formatos de datos como JSON o YAML (`type: 'data'`). También incluye un `schema` que define la forma de tu frontmatter o datos de entrada.
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 4]
```diff
@@ -0,0 +1,753 @@
...
+3. **Exporta un único objeto `collections`** para registrar tus colecciones.
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:147]:
```suggestion
3. **Exportar un único objeto `collections`** para registrar tus colecciones. 
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 5]
```diff
@@ -0,0 +1,753 @@
...
+
+Para definir tu primer colleción, crea un archivo `src/content/config.ts` si no existe (las extensiones `.js` y `.mjs` también son compatibles). Este archivo debe:
+
+1. **Importa las utilidades adecuadas** de `astro:content`.
+2. **Define cada colección que deseas validar.** Esto incluye un `type` (introducido en Astro v2.5.0) que especifica si la colección contiene formatos de autoría de contenido como Markdown (`type: 'content'`) o formatos de datos como JSON o YAML (`type: 'data'`). También incluye un `schema` que define la forma de tu frontmatter o datos de entrada.
+3. **Exporta un único objeto `collections`** para registrar tus colecciones. 
+
+```ts
+// src/content/config.ts
+// 1. Importa las utilidades de `astro:content`
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:151]:
```suggestion
// 1. Importar las utilidades de `astro:content`
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 6]
```diff
@@ -0,0 +1,753 @@
...
+import { z, defineCollection } from 'astro:content';
+
+// 2. Define un `type` y `schema` para cada colección
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:154]:
```suggestion
// 2. Definir un `type` y `schema` para cada colección
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 7]
```diff
@@ -0,0 +1,753 @@
...
+const blogCollection = defineCollection({
+  type: 'content', // v2.5.0 y posteriores
+  schema: z.object({
+    title: z.string(),
+    tags: z.array(z.string()),
+    image: z.string().optional(),
+  }),
+});
+
+// 3. Exporta un único objeto `collections` para registrar tu(s) colección(es)
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:164]:
```suggestion
// 3. Exportar un único objeto `collections` para registrar tu(s) colección(es)
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 8]
```diff
@@ -0,0 +1,753 @@
...
+});
+
+export const collections = {
+  'blog': blogCollection,
+  'newsletter': newsletter,
+  'authors': authors,
+};
+```
+
+Como tu proyecto crece, también eres libre de reorganizar tu base de código y mover la lógica fuera del archivo `src/content/config.ts`. Definir tus esquemas por separado puede ser útil para reutilizar esquemas en múltiples colecciones y compartir esquemas con otras partes de tu proyecto.
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:196]:
```suggestion
A medida que tu proyecto crece, también eres libre de reorganizar tu base de código y mover la lógica fuera del archivo `src/content/config.ts`. Definir tus esquemas por separado puede ser útil para reutilizar esquemas en múltiples colecciones y compartir esquemas con otras partes de tu proyecto.
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 9]
```diff
@@ -0,0 +1,753 @@
...
+
+```ts
+// src/content/config.ts
+// 1. Importa tus utilidades y esquemas
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:200]:
```suggestion
// 1. Importar tus utilidades y esquemas
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 10]
```diff
@@ -0,0 +1,753 @@
...
+const blogCollection = defineCollection({
+  type: 'content',
+  schema: blogSchema,
+});
+const authorCollection = defineCollection({
+  type: 'data',
+  schema: authorSchema,
+});
+
+// 3. Exporta múltiples colecciones para registrarlas
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:214]:
```suggestion
// 3. Exportar múltiples colecciones para registrarlas
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 11]
```diff
@@ -0,0 +1,753 @@
...
+export const collections = {
+  'blog': blogCollection,
+  'authors': authorCollection,
+};
+```
+
+### Usando esquemas de colección de terceros
+
+Puedes importar esquemas de colección desde cualquier lugar, incluidos los paquetes npm externos. Esto puede ser útil cuando trabajas con temas y bibliotecas que proporcionan sus propios esquemas de colección para que los uses.
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:223]:
```suggestion
Puedes importar esquemas de colección desde cualquier lugar, incluyendo paquetes npm externos. Esto puede ser útil cuando trabajas con temas y bibliotecas que proporcionan sus propios esquemas de colección para que los uses.
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 12]
```diff
@@ -0,0 +1,753 @@
...
+
+
+```ts
+// src/content/config.ts
+import { blogSchema } from 'my-blog-theme';
+const blogCollection = defineCollection({ type: 'content', schema: blogSchema });
+
+// Exporta la colección de blog, usando un esquema externo de 'my-blog-theme'
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:231]:
```suggestion
// Exportar la colección de blog, usando un esquema externo de 'my-blog-theme'
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 13]
```diff
@@ -0,0 +1,753 @@
...
+export const collections = {
+  'blog': blogCollection,
+};
+```
+
+
+### Definiendo tipos de datos con Zod
+
+Astro usa [Zod](https://github.com/colinhacks/zod) para alimentar sus esquemas de contenido. Con Zod, Astro puede validar el frontmatter de cada archivo dentro de una colección *y* proporcionar tipos automáticos de TypeScript cuando consultas el contenido desde dentro de tu proyecto.
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:240]:
```suggestion
Astro usa [Zod](https://github.com/colinhacks/zod) para potenciar sus esquemas de contenido. Con Zod, Astro puede validar el frontmatter de cada archivo dentro de una colección *y* proporcionar tipos automáticos de TypeScript cuando consultas el contenido desde dentro de tu proyecto.
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 14]
```diff
@@ -0,0 +1,753 @@
...
+
+```js
+import { getCollection, getEntry } from 'astro:content';
+
+// Obtén todas las entradas de una colección.
+// Requiere el nombre de la colección como argumento.
+// Ejemplo: recupera `src/content/blog/**`
+const allBlogPosts = await getCollection('blog');
+
+// Obtén todas las entradas de una colección.
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:346]:
```suggestion
// Obtén una única entrada de colleción.
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 15]
```diff
@@ -0,0 +1,753 @@
...
+// Requiere el nombre de la colección y también
+// el `slug` de la entrada (colecciones de contenido) o `id` (colecciones de datos) 
+// Ejemplo: recupera `src/content/authors/grace-hopper.json`
+const graceHopperProfile = await getEntry('authors', 'grace-hopper');
+```
+
+Both functions return content entries as defined by the [`CollectionEntry`](/es/reference/api-reference/#colección-de-tipo-entrada) type.
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:353]: falta traducir

File: src/content/docs/es/guides/content-collections.mdx [hunk 16]
```diff
@@ -0,0 +1,753 @@
...
+Ambos
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:354]: eliminar esto

File: src/content/docs/es/guides/content-collections.mdx [hunk 17]
```diff
@@ -0,0 +1,753 @@
...
+
+```astro title="src/pages/blog/welcome.astro"
+---
+import { getEntry, getEntries } from 'astro:content';
+
+const blogPost = await getEntry('blog', 'welcome');
+
+// Resuelve una referencia singular
+const author = await getEntry(blogPost.data.author);
+// Resuelve un arreglo de referencias
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:368]:
```suggestion
// Resuelve un array de referencias
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 18]
```diff
@@ -0,0 +1,753 @@
...
+// Ejemplo: Filtra las entradas por subdirectorio en la colección
+import { getCollection } from 'astro:content';
+const englishDocsEntries = await getCollection('docs', ({ id }) => {
+  return id.startsWith('en/');
+});
+```
+
+### Usando contenido en plantillas de Astro
+
+Una vez que hayas consultado tus entradas de colección, puedes acceder a cada entrada directamente dentro de la plantilla de tu componente Astro. Esto te permite renderizar HTML para cosas como enlaces a tu contenido (usando el `slug` de contenido) o información sobre tu contenido (usando la propiedad `data`).
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:413]:
```suggestion
Una vez que hayas consultado tus entradas de colección, puedes acceder a cada entrada directamente dentro de la plantilla de tu componente de Astro. Esto te permite renderizar HTML para cosas como enlaces a tu contenido (usando el `slug` de contenido) o información sobre tu contenido (usando la propiedad `data`).
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 19]
```diff
@@ -0,0 +1,753 @@
...
+    </li>
+  ))}
+</ul>
+```
+
+### Pasando contenido como props
+
+Un componente también puede pasar un contenido completo como una prop.
+
+Si haces esto, puedes usar la utilidad [`CollectionEntry`](/es/reference/api-reference/#colección-de-tipo-entrada) para escribir correctamente las props de tu componente usando TypeScript. Esta utilidad toma un argumento de string que coincide con el nombre del esquema de tu colección y heredará todas las propiedades de ese esquema de colección.
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:439]:
```suggestion
Si haces esto, puedes usar la utilidad [`CollectionEntry`](/es/reference/api-reference/#colección-de-tipo-entrada) para tipar correctamente las props de tu componente usando TypeScript. Esta utilidad toma un argumento de tipo string que coincide con el nombre del esquema de tu colección y heredará todas las propiedades de ese esquema de colección.
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 20]
```diff
@@ -0,0 +1,753 @@
...
+
+```astro /CollectionEntry(?:<.+>)?/
+---
+// src/components/BlogCard.astro
+import type { CollectionEntry } from 'astro:content';
+interface Props {
+  post: CollectionEntry<'blog'>;
+}
+
+// `post` will match your 'blog' collection schema type
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:449]: falta traducir

File: src/content/docs/es/guides/content-collections.mdx [hunk 21]
```diff
@@ -0,0 +1,753 @@
...
+const { Content } = await entry.render();
+---
+<h1>{entry.data.title}</h1>
+<Content />
+```
+
+Esto generará una nueva página para cada entrada en la colección `blog`. Por ejemplo, una entrada en `src/content/blog/hello-world.md` tendrá un slug de `hello-world`, y por lo tanto su URL final será `/posts/hello-world/`.
+
+:::note
+Si tu slugs personalizados contienen el carácter `/` para producir URL con múltiples segmentos de ruta, debes usar un [parámetro rest (`[...slug]`)](/es/core-concepts/routing/#parámetros-rest) en el nombre de archivo `.astro` para esta página de enrutamiento dinámico.
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:504]:
```suggestion
Si tus slugs personalizados contienen el carácter `/` para producir URLs con múltiples segmentos de ruta, debes usar un [parámetro rest (`[...slug]`)](/es/core-concepts/routing/#parámetros-rest) en el nombre de archivo `.astro` para esta página de enrutamiento dinámico.
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 22]
```diff
@@ -0,0 +1,753 @@
...
+
+
+```astro
+---
+// src/pages/posts/[...slug].astro
+import { getEntry } from "astro:content";
+// 1. Obtén el slug de la solicitud entrante del servidor
+const { slug } = Astro.params;
+if (slug === undefined) {
+	throw new Error("Slug is required");
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:519]:
```suggestion
	throw new Error("Slug es requerido");
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 23]
```diff
@@ -0,0 +1,753 @@
...
+
+    El índice del blog en el tutorial enumera una tarjeta para cada publicación. Esto se convierte en:
+
+    ```astro title="src/pages/blog.astro" "post.data" "getCollection(\"posts\")" "'/posts/' + post.slug"
+    ---
+    import { getCollection } from "astro:content";
+    import BaseLayout from "../layouts/BaseLayout.astro";
+    import BlogPost from "../components/BlogPost.astro";
+
+    const pageTitle = "My Astro Learning Blog";
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:623]:
```suggestion
    const pageTitle = "Mi blog de aprendizaje de Astro";
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 24]
```diff
@@ -0,0 +1,753 @@
...
+
+    Previamente, `pubDate` era un string. Ahora, después de introducir tipos para el frontmatter de tus publicaciones, `pubDate` es un `Date`.
+    Para renderizar la fecha, conviértela en un string:
+
+    ```astro title="src/layouts/MarkdownPostLayout.astro" "frontmatter.pubDate.toDateString()"
+    ...
+    <BaseLayout pageTitle={frontmatter.title}>
+      <p>{frontmatter.pubDate.toDateString()}</p>
+      <p><em>{frontmatter.description}</em></p>
+      <p>Written by: {frontmatter.author}</p>
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:701]:
```suggestion
      <p>Escrito por: {frontmatter.author}</p>
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 25]
```diff
@@ -0,0 +1,753 @@
...
+    Por último, el tutorial de proyecto de blog incluye un feed RSS. Esta función también debe usar `getCollection` y el objeto `data`, y convertirse en una función asíncrona para hacerlo:
+
+    ```js title="src/pages/rss.xml.js" {4-5, 10-15} 
+    import rss from "@astrojs/rss";
+    import { getCollection } from "astro:content";
+
+    export async function get() {
+      const posts = await getCollection('posts');
+      return rss({
+        title: 'Astro Learner | Blog',
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:716]:
```suggestion
        title: 'Estudiante de Astro | Blog',
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 26]
```diff
@@ -0,0 +1,753 @@
...
+        description: 'My journey learning Astro',
```
Comment (Waxer59) [inline src/content/docs/es/guides/content-collections.mdx:717]:
```suggestion
        description: 'Mi viaje aprendiendo Astro',
```

File: src/content/docs/es/guides/content-collections.mdx [hunk 27]
```diff
@@ -0,0 +1,753 @@
...
+```ts
+import { defineCollection, reference, z } from 'astro:content';
+
+const blog = defineCollection({
+  type: 'content',
+  schema: z.object({
+    title: z.string(),
+    // Referencia a un único autor de la colección `authors` por `id`
+    author: reference('authors'),
+    // Referencia a una matriz de publicaciones relacionadas de la colección `blog` por `slug`
```
[the remark this answers is not in the document]
Comment (dreyfus92) [reply src/content/docs/es/guides/content-collections.mdx:292]: Aquí lo voy a cambiar a `arreglo`

File: src/content/docs/es/guides/content-collections.mdx (hunk 17 above)
Comment (dreyfus92) [reply src/content/docs/es/guides/content-collections.mdx:368]: Aquí igual, voy a dejar la palabra `arreglo`

**Review by Waxer59 — APPROVED** [review]
LGTM ✔

File: src/content/docs/es/guides/content-collections.mdx [hunk 28]
```diff
@@ -0,0 +1,751 @@
...
+```astro "{ remarkPluginFrontmatter }"
+---
+import { getEntry } from 'astro:content';
+const blogPost = await getEntry('blog', 'post-1');
+const { remarkPluginFrontmatter } = await blogPost.render();
+---
+<p>{blogPost.data.title} — {remarkPluginFrontmatter.readingTime}</p>
+```
+
+<RecipeLinks slugs={["en/recipes/reading-time" ]}/>
```
Comment (Yan-Thomas) [inline src/content/docs/es/guides/content-collections.mdx:748]:
```suggestion
<RecipeLinks slugs={["es/recipes/reading-time" ]}/>
```

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks peeps 🚀

**Pull request merged**